libcaf_core/test/detail/config_consumer.cpp

```diff
@@ -0,0 +1,112 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#define CAF_SUITE detail.config_consumer
+
+#include "caf/detail/config_consumer.hpp"
+
+#include "caf/test/dsl.hpp"
+
+#include "caf/detail/parser/read_config.hpp"
+
+using std::string;
+
+using namespace caf;
+
+// List-of-strings.
+using ls = std::vector<std::string>;
+
+namespace {
+
+constexpr const string_view test_config1 = R"(
+is_server=true
+port=4242
+nodes=["sun", "venus", ]
+logger{
+  file-name = "foobar.conf" # our file name
+}
+scheduler { # more settings
+  timing  =  2us # using microsecond resolution
+}
+)";
+
+constexpr const string_view test_config2 = R"(
+is_server = true
+logger = {
+  file-name = "foobar.conf"
+}
+port = 4242
+scheduler : {
+  timing = 2us,
+}
+nodes = ["sun", "venus"]
+)";
+
+struct fixture {
+  config_option_set options;
+  settings config;
+
+  fixture() {
+    options.add<bool>("global", "is_server", "enables server mode")
+      .add<uint16_t>("global", "port", "sets local or remote port")
+      .add<ls>("global", "nodes", "list of remote nodes")
+      .add<string>("logger", "file-name", "log output file")
+      .add<int>("scheduler", "padding", "some integer")
+      .add<timespan>("scheduler", "timing", "some timespan");
+  }
+};
+
+} // namespace
+
+CAF_TEST_FIXTURE_SCOPE(config_consumer_tests, fixture)
+
+CAF_TEST(config_consumer) {
+  string_view str = test_config1;
+  detail::config_consumer consumer{options, config};
+  string_parser_state res{str.begin(), str.end()};
+  detail::parser::read_config(res, consumer);
+  CAF_CHECK_EQUAL(res.code, pec::success);
+  CAF_CHECK_EQUAL(string_view(res.i, res.e), string_view());
```

Comment:
```suggestion
  CAF_CHECK_EQUAL(res.i, res.e);
```

Reply:
These two checks give the same yes/no answer, but there's a reason I've written it like this: writing it this way will print what's left in the buffer. This means, if the check fails, it also gives you the information you need to hunt for the bug right away. 🙂

Reply:
Neat.